chrome/content_script.js

```diff
@@ -442,14 +452,18 @@ passwordalert.completePageInitializationIfReady_ = function() {
       // Delete any previously considered password in case this is a re-prompt
       // when an incorrect password is entered, such as a ServiceLoginAuth page.
       chrome.runtime.sendMessage({action: 'deletePossiblePassword'});
-      var loginForm = document.getElementById('gaia_loginform');
-      // The chooser is also a gaia_loginform, so verify we're on a password
-      // entry page.
+      var loginForm = document.querySelector('#view_container > form');
+      // The chooser is no longer a gaia_loginform, so verify we're on a password
+      // entry page by finding a form in a view_container.
       if (loginForm && document.getElementById('Email')) {
         loginForm.addEventListener(
             'submit', passwordalert.saveGaiaPassword_, true);
       } else if (document.getElementById('hiddenEmail') &&
                  document.getElementsByName('password')){
+        document.getElementById("passwordNext").addEventListener('click',function (){
```

Comment:
Whatever you think would be best here. You're right that it won't trigger on pressing enter, thanks for the catch. We just needed some way to store the hidden email and password when the user took action on the next button. Otherwise these elements don't exist once the user reaches the two factor screen.

Reply:
In my testing, we do not need the temp_user_ and temp_pass_. The window.onbeforeunload handler (saveGaia2Password_) can still see document.getElementById('hiddenEmail') and document.getElementsByName('password'). This makes it so that we do not need to have the click handler and that onbeforeunload will handle all navigation, including pressing enter.

If I'm missing a case where the temp_user_ is needed, please let me know.